### PR TITLE
coding demo views of Label, texfield, button

### PR DESCRIPTION
**Pull request by duongcuong4395**

### CODE REVIEW

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+//  Copyright © 2021 Duong Cuong. All rights reserved.
+//
+
+import Foundation
+
+
+import SnapKit
+import UIKit
+
+extension UIButton {
```
Comment (ReverseThatApp) [inline Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:15]: we better create a new file and put the extension methods there, for example UIButtonExtension.swift

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    func title_With(title: String, color: UIColor, state: UIControl.State) {
```
Comment (ReverseThatApp) [inline Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:16]: please follow method name convention using camel case, for example, func titleWith(_ title: String...)

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    
+    func setup_Button() {
+        setup_Button_Constrains()
+        
+
```
Comment (ReverseThatApp) [inline Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:79]: please remove empty line if not needed

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        case .action:
+            setup_Button_Action()
+        case .shadow:
+            setup_Button_Shadow()
+        case .image:
+            setup_Button_Image()
+        }
+    }
+    
+    func setup_Button_default() {
```
Comment (ReverseThatApp) [inline Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:141]: if this method is suppose to be private (only been used in this class only), then consider to add ```private``` modifier: ```private func setupButtonDefault() { ...}```

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    
+    func shadow_With(color: CGColor, offset: CGSize, opacity: Float, radius: CGFloat) {
+        let button = self // changes made here
+        button.layer.shadowColor = color
+        button.layer.shadowOffset = offset
+        button.layer.shadowOpacity = opacity
+        button.layer.shadowRadius = radius
+    }
+    
+    func image_With(img: UIImage, control: UIControl.State)  {
```
Comment (ReverseThatApp) [inline Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:36]: actually we dont need this method, you just can set image directly using built-in ```setImage``` method

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+
+import SnapKit
+import UIKit
+
+extension UIButton {
+    func title_With(title: String, color: UIColor, state: UIControl.State) {
+        let button = self // changes made here
+        button.setTitle(title, for: state)
+        button.setTitleColor(.black, for: state)
```
Comment (ReverseThatApp) [inline Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:19]: why not using ```color``` parameter but hard coding ```.black``` here?

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift (hunk 2 above)
Comment (ReverseThatApp) [inline Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:16]: please follow method naming convention using camel case, for example: ```func titleWith(...```

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift (hunk 5 above)
Comment (ReverseThatApp) [inline Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:36]: this extension method seems doing nothing, please remove it

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift (hunk 4 above)
Comment (duongcuong4395) [reply Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:141]: If I just write: func funcName(),  the program will understand the default as private method?
Is this right?

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift (hunk 3 above)
Comment (duongcuong4395) [reply Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:79]: Can you tell me why it's not necessary? And please give me some ideas for this problem.

Comment (ReverseThatApp) [reply Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:79]: this has 2 empty lines, if you want to make empty line just keep 1 empty line is enough

File: Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift (hunk 4 above)
Comment (ReverseThatApp) [reply Swift_Tutor/Swift_Tutor/UIElements/Button/ButtonDemoView.swift:141]: no, if you want to make the method name private, please use ```private``` modifier, default is ```internal``` which is public inside that module, google swift modifier for more details

**Pull request closed without merging**